### ISSUE TITLE
Precompilation crash? on OS X 10.11.2

### ISSUE THREAD

**Issue by 315234** [stevengj/Cubature.jl#14]
```
$ julia --version
julia version 0.4.2

$ julia
julia> using Cubature
Assertion failed: (ctx->gensym_assigned.at(idx)), function emit_expr, file codegen.cpp, line 3136.

signal (6): Abort trap: 6
__pthread_kill at /usr/lib/system/libsystem_kernel.dylib (unknown line)
[1]    78750 abort      julia

$ uname -a
Darwin <hostname> 15.2.0 Darwin Kernel Version 15.2.0; root:xnu-3248.20.55~2/RELEASE_X86_64 x86_64
```

i.e. This is on OS X 10.11.2 (15C50). Perhaps this is a Julia bug, in which case I will report it upstream.

**Comment by stevengj** [comment]
I can't reproduce.  Can you try `Pkg.update(); Pkg.build("Cubature")`?

**Comment by jerjorg** [comment]
I have the same problem with Julia v0.4.3 and OSX 10.11.3

```
julia> Pkg.update()
INFO: Updating METADATA...
INFO: Updating ksi...
INFO: Computing changes...
INFO: No packages to install, update or remove

julia> Pkg.build("Cubature")
INFO: Building Cubature
INFO: Cubature 1.0.2 is already installed.

julia> using Cubature
Assertion failed: (ctx->gensym_assigned.at(idx)), function emit_expr, file codegen.cpp, line 3138.

signal (6): Abort trap: 6
__pthread_kill at /usr/lib/system/libsystem_kernel.dylib (unknown line)
Abort trap: 6
Jer:Codes jeremy$ 
```

**Comment by stevengj** [comment]
I just tried with Julia 0.4.3 and OSX 10.11.3, and it still works fine for me:
```
julia> Pkg.build("Cubature")
INFO: Building Cubature
INFO: Cubature 1.0.2 is already installed.

julia> using Cubature
INFO: Recompiling stale cache file /Users/stevenj/.julia/lib/v0.4/Cubature.ji for module Cubature.
```

**Comment by 315234** [comment]
Still happening for me, even with a completely clean install.

    $ uname -a
    Darwin www.me.com 15.3.0 Darwin Kernel Version 15.3.0: Thu Dec 10 18:40:58 PST 2015; root:xnu-3248.30.4~1/RELEASE_X86_64 x86_64
    
    $ mv .julia .julia_hidden
    
    $ julia
    
    julia> println(VERSION)
    0.4.3
    
    julia> Pkg.add("Cubature")
    INFO: Initializing package repository /Users/Me/.julia/v0.4
    INFO: Cloning METADATA from git://github.com/JuliaLang/METADATA.jl
    INFO: Cloning cache of BinDeps from git://github.com/JuliaLang/BinDeps.jl.git
    INFO: Cloning cache of Compat from git://github.com/JuliaLang/Compat.jl.git
    INFO: Cloning cache of Cubature from git://github.com/stevengj/Cubature.jl.git
    INFO: Cloning cache of SHA from git://github.com/staticfloat/SHA.jl.git
    INFO: Cloning cache of URIParser from git://github.com/JuliaWeb/URIParser.jl.git
    INFO: Installing BinDeps v0.3.21
    INFO: Installing Compat v0.7.12
    INFO: Installing Cubature v1.0.10
    INFO: Installing SHA v0.1.2
    INFO: Installing URIParser v0.1.2
    INFO: Building Cubature
    INFO: Installing Cubature 1.0.2 library...
    INFO: Package database updated
    
    julia> using Cubature
    INFO: Precompiling module Cubature...
    Assertion failed: (ctx->gensym_assigned.at(idx)), function emit_expr, file codegen.cpp, line 3138.
    
    signal (6): Abort trap: 6
    __pthread_kill at /usr/lib/system/libsystem_kernel.dylib (unknown line)
    [1]    33757 abort      julia

**Comment by 315234** [comment]
Created a new user on the same system, repeated the above steps and got the same crash.

**Comment by stevengj** [comment]
Weird... Since it is an assertion failure, I'm inclined to say it is a Julia bug and should be filed as a `julia` issue.   May be hard to track down unless a developer can reproduce it, though.

**Comment by randy3k** [comment]
A similar issue was also reported on RCall.jl https://github.com/JuliaStats/RCall.jl/issues/131#issuecomment-230122191 

@jerjorg @315234  Did you install julia from homebrew? My experience tells me that homebrew Julia does not work very well with `ccall`.

**Comment by 315234** [comment]
Yes, I installed from homebrew.

**Comment by stevengj** [comment]
This is JuliaLang/julia#16546.  It has been fixed in 0.5, but is apparently still waiting for someone to backport the patch to Julia 0.4.
